File: 1984.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
  int minimumDifference(vector<int> &nums, int k) {
    int minVal = __INT_MAX__;
    sort(nums.begin(), nums.end());
    for (int l = 0; l < nums.size() - k + 1; ++l) {
      minVal = min(minVal, nums[l + k - 1] - nums[l]);
    }
    return minVal;
  }
};

File: 1984.py
from sys import maxsize


class Solution:
    def minimumDifference(self, nums: list[int], k: int) -> int:
        nums.sort()
        minDiff = maxsize
        for l in range(len(nums) - k + 1):
            minDiff = min(nums[l + k - 1] - nums[l], minDiff)
        return minDiff


assert Solution().minimumDifference([9, 4, 1, 7], 2) == 2
